packages/p2p/src/components/my-ads/quick-add-modal.jsx

```diff
@@ -73,9 +73,23 @@ const QuickAddModal = ({ advert }) => {
         }
     };
 
+    const hideModal = close_all => {
+        setSelectedMethods([]);
+        if (!close_all) {
+            if (my_ads_store.should_show_add_payment_method) {
+                my_ads_store.setShouldShowAddPaymentMethod(false);
+            } else {
+                my_ads_store.hideQuickAddModal();
+            }
+        } else {
+            my_ads_store.setShouldShowAddPaymentMethod(false);
+            my_ads_store.hideQuickAddModal();
+        }
+    };
+
     React.useEffect(() => {
         setSelectedMethods([]);
-        my_ads_store.setShouldShowAddPaymentMethod(!my_profile_store?.advertiser_has_payment_methods);
+        my_ads_store.setShouldShowAddPaymentMethod(false);
 
```

Comment:
Why are we setting this to false instead of tracking an observable that tells us this is false

Reply:
From the design requirements, when user clicks the `+ Add` and the `quick-add-modal` component is mounted, the design expects to show the list of payment method cards instead of the add payment method form even though the user has not added payment methods, as specified by the observable `my_profile_store.advertiser_has_payment_methods`. This was replaced with `false` (to ensure that the payment method cards would always be displayed first) since if the user does not have any payment methods, the observable `advertiser_has_payment_methods` would be true and it would display the `AddPaymentMethod` form, which by design it should be displaying the payment method add card.

*When user clicks on +Add, it should show the payment method add card instead of the form first*
![Screenshot 2022-05-11 at 12 59 37 PM](https://user-images.githubusercontent.com/103016120/167779309-ba01cb1d-4df6-4c69-b7ce-d52c113e92c1.png)


*If `setShouldShowAddPaymentMethod` is set to the observable `advertiser_has_payment_methods`, it would always show the add payment method form first if the user has not added payment methods*
![Screenshot 2022-05-11 at 1 00 27 PM](https://user-images.githubusercontent.com/103016120/167772535-9a6a15a4-e831-47ac-8efb-7b9beb09ca35.png)

Reply:
The flow:
![Screenshot 2022-05-11 at 2 10 14 PM](https://user-images.githubusercontent.com/103016120/167780534-5e59ea49-0733-46c3-b964-4f352fef61f2.png)